### PR TITLE
Creating generic positions url

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Create Generic Positions url

Dual Merge: 
- [FE PR ](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2246)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/295)

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
Back in WIP to update EP from WS call

File: talentmap_api/common/common_helpers.py [hunk 1]
```diff
@@ -179,18 +179,20 @@ def ensure_date(date, utc_offset=0):
     '''
     try:
         if not date:
+            logger.warn(f"Invalid date: Date parameter must be a date object or string. date: {date}")
```
Comment (scott062) [inline talentmap_api/common/common_helpers.py:182]: This is saying, no date arg was ever given right?

File: talentmap_api/common/common_helpers.py [hunk 2]
```diff
@@ -179,18 +179,20 @@ def ensure_date(date, utc_offset=0):
...
             return None
         elif isinstance(date, str):
             return parser.parse(date).astimezone(datetime.timezone.utc) - datetime.timedelta(hours=utc_offset)
         elif isinstance(date, datetime.date):
             return date.astimezone(datetime.timezone(datetime.timedelta(hours=utc_offset)))
         else:
-            logger.warn(f"Parameter {date} must be a date object or string")
-            return None
+            logger.warn(f"Date parameter must be a date object or string. date: {date}")
```
[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply talentmap_api/common/common_helpers.py:189]: oh, i like that!

File: talentmap_api/common/common_helpers.py (hunk 1 above)
Comment (elizabeth-jimenez) [reply talentmap_api/common/common_helpers.py:182]: yep, `None`, `''`, etc

**Comment by elizabeth-jimenez** [comment]
@scott062 multi-line logs are apparently bad practice, so that's why i split them out

File: talentmap_api/common/common_helpers.py [hunk 3]
```diff
@@ -179,18 +179,26 @@ def ensure_date(date, utc_offset=0):
...
+            logger.warn(f"date: {date}")
+            logger.warn(f"type(date): {type(date)}")
+            return "Invalid date"
     except ValueError:
         try:
             return parser.parse(date + '.000Z').astimezone(datetime.timezone.utc) - datetime.timedelta(hours=utc_offset)
         except:
+            logger.warn(f"Invalid date: Date parameter must be a date object or string.")
+            logger.warn(f"date: {date}")
+            logger.warn(f"type(date): {type(date)}")
```
Comment (scott062) [inline talentmap_api/common/common_helpers.py:201]: 🛑 V nice

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,13 +1,3 @@
...
-- [Mock PR](linkToMockPR)
-
-
----
-
-### Pull Request Checklist
-
-- [ ] Postman tests for any new endpoints
+- [FE PR](linkToFrontEndPR)
+- [Mock PR](linkToMockPR)
```
Comment (scott062) [inline .github/pull_request_template.md:3]: 😿 🪄 🧙 💀 ⚰️

File: docker-compose-api-only.yml [hunk 5]
```diff
@@ -40,6 +40,7 @@ services:
       - PANEL_API_URL=http://host.docker.internal:3333/v1/panels
       - PERSON_API_URL=http://host.docker.internal:3333/v3/persons
       - BIDS_API_V2_URL=http://host.docker.internal:3333/v2/bids
+      - POSITIONS_API_V2_URL=http://host.docker.internal:3333/v2/positions
```
Comment (scott062) [inline docker-compose-api-only.yml:43]: I don't know the best way to do this, but we need to consolidate release env variables changes somehow, don't know if we can just add a label or flag to this PR to remind us. Food for thought - no changes needed

File: .github/pull_request_template.md (hunk 4 above)
Comment (elizabeth-jimenez) [reply .github/pull_request_template.md:3]: 🪄 🧙‍♂️ 😭

File: talentmap_api/fsbid/services/positions.py [hunk 6]
```diff
@@ -25,6 +27,24 @@ def get_position(id, jwt_token):
 
     return pydash.get(position, 'results[0]') or None
 
+def get_positions(query, jwt_token):
+    '''
+    Gets unavailable positions
```
Comment (scott062) [inline talentmap_api/fsbid/services/positions.py:32]: Technically gets ANY position - not to be confused with getting cycle positions in unavailable status. May want to change comment to clarify

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/positions.py]: updated to generic

**Review by scott062 — APPROVED** [review]
👩‍🎤

**Pull request merged**